ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Util/Functors.hs

```diff
@@ -1,8 +1,24 @@
-{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE PolyKinds                  #-}
+{-# LANGUAGE StandaloneKindSignatures   #-}
 
-module Ouroboros.Consensus.HardFork.Combinator.Util.Functors (Product2 (..)) where
+module Ouroboros.Consensus.HardFork.Combinator.Util.Functors (
+    Flip (..)
+  , K1 (..)
+  , Product2 (..)
+  ) where
 
+import           Data.Kind (Type)
 import           GHC.Generics (Generic)
+import           NoThunks.Class (NoThunks)
 
 data Product2 f g x y = Pair2 (f x y) (g x y)
   deriving (Eq, Generic, Show)
+
+type Flip :: (k -> l -> Type) -> l -> k -> Type
```

Comment:
Why do we use `k` and `l` instead of `a` and `b` or `x` and `y`. Later we have the `K1` type, which overloads this letter :)